### ISSUE TITLE
api_logfile setting takes no effect

### ISSUE THREAD

**Issue by tyeapple** [saltstack/salt#38479]
source code: salt/config/__init__.py, line: 3435 - 3448
User can't customize api_logfile.

Version: tags/2016.11.1

```python
def api_config(path):
    '''
    Read in the Salt Master config file and add additional configs that
    need to be stubbed out for salt-api
    '''
    # Let's grab a copy of salt's master opts
    opts = client_config(path, defaults=DEFAULT_MASTER_OPTS)
    # Let's override them with salt-api's required defaults
    api_opts = DEFAULT_API_OPTS
    api_opts.update({
        'pidfile': opts.get('api_pidfile', DEFAULT_API_OPTS['api_pidfile']),
    })
    opts.update(api_opts)
    return opts
```

**Comment by Ch3LL** [comment]
@tyeapple mind giving #38560 a try?

**Comment by rallytime** [comment]
Since we didn't receive a response here, and this should be all fixed up by the changes made in #38560, #38527, and #38585 (the last one containing tests for this issue), I'm going to close this. If this pops up again, we are happy to re-open the issue and take another look.
